sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -3415,10 +3415,10 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
     val specsAndLocs = ctx.partitionSpecLocation.asScala.map { splCtx =>
       val spec = visitNonOptionalPartitionSpec(splCtx.partitionSpec)
       val location = Option(splCtx.locationSpec).map(visitLocationSpec)
-      spec -> location
+      UnresolvedPartitionSpec(spec, location)
     }
-    AlterTableAddPartitionStatement(
-      visitMultipartIdentifier(ctx.multipartIdentifier),
+    AlterTableAddPartition(
+      UnresolvedTableOrView(visitMultipartIdentifier(ctx.multipartIdentifier)),
```

Comment:
this doesn't support view, we should use `UnresolvedTable`

Reply:
We shouldn't change behavior in this PR. If `AlterTableAddPartition` fails before if the name refers to a view, we should use `UnresolvedTable` to keep the failing behavior.

Reply:
ok